### ISSUE TITLE
Add a color theme switcher

### ISSUE THREAD

**Issue by ericwbailey** [a11yproject/a11yproject.com#1008]
This issue is a combination of https://github.com/a11yproject/a11yproject.com/issues/964 and https://github.com/a11yproject/a11yproject.com/issues/939. We're proposing introducing a color theme switcher component to deal with their concerns.

## Current issues

- Some color combinations have been reported to be too harsh
    - Notably, white on bright pink is causing Chromostereopsis-style effect
- A dark mode can be helpful for things like low light/vision scenarios

## Goals for the color theme switcher

- [x] An accessible, semantic toggle for the control
- [x] The current theme
- [ ] A muted theme
- [x] A dark mode theme, triggered by `prefers-color-scheme: dark` but capable of being updated via the switcher
- [x] Styling that integrates into our existing Sass setup, and not a filter overlay
- [x] Localstorage to remember a user's preference

## Relevant information

Max Böck's implementation: 
- https://mxb.dev/blog/color-theme-switcher/
- https://github.com/maxboeck/mxb/blob/master/src/includes/themepicker.njk

**Additional examples shared via #939**

- https://hankchizljaw.com/wrote/create-a-user-controlled-dark-or-light-mode
- https://css-tricks.com/a-complete-guide-to-dark-mode-on-the-web/

Tagging @dennisgaebel, as they were involved in the discussions.

**Comment by dennisgaebel** [comment]
@ericwbailey Woot! I'm thinking it would be best to have Tatiana whip up some comps with colors to use if that's at all possible? LMK 🤝 

Here's a list adding on to yours above I think we need comps for in terms of color variants. Any of these color schemes should pass [WCAG standards](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#contrast-minimum) of course.

**Color Schemes**
1. Default (current theme)
2. Muted (I think a better term might be Bright? or even Desaturated?)
3. Dark Mode
4. Protanopia
5. Deuteranopia
6. Tritanopia

Here's what I made to help us preview all this using InVision Boards. I also included some rough draft ideas for dark mode I made via DevTools https://projects.invisionapp.com/boards/54407L6BF8Q Should help give you an idea of where I'm going for this work.

**Comment by StommePoes** [comment]
I mean, imagine if the insufficient-contrast hot pink were adjusted to #cd005a, which also gives 5.6:1 contrast ratio.

I'm not against having a buttload of themes, but I would be disappointed that the main/default theme on an accessibility site has contrast under 3:1 for text and indications of things. (Technically, offering a theme with sufficient contrast passes WCAG so this isn't a WCAG complaint). This site is not just a resource for us to read, but visually is a showcase of what you can do design-wise and be accessible... low-contrast colours presented by default kinda goes against that, and I think the general colour scheme can remain if the colours are at least tweaked to hit the not-difficult goal of 3:1. It would still look fairly pink. The colour I posted above is much much darker, but imagine you lightened it to a 3:1 contrast, it'd be even closer to the original colours.

**Comment by dennisgaebel** [comment]
@StommePoes 

> …but I would be disappointed that the main/default theme on an accessibility site has contrast under 3:1 for text and indications of things. (Technically, offering a theme with sufficient contrast passes WCAG so this isn't a WCAG complaint)

I agree that the current site should meet standards. Right now it's more of a "do as i say not as  I do" message to people IMHO. Currently the site contains 20 color variations listed below:

![Screenshot 2020-07-21 23 38 21](https://user-images.githubusercontent.com/13485205/88414859-89030680-cdab-11ea-832f-2918132a20f7.png)

> I'm not against having a buttload of themes…

I still feel strongly that the current color scheme should be adjusted to meet standards without sacrificing on visual design goals. Ultimately a "default" theme should be passing WCAG guidelines. Only other themes that should be added is "dark mode" and themes related to color blind modes. Having a good default theme would eliminate the need to have a muted theme IMO

**Comment by rachel-fischoff** [comment]
Hi! Just checking in to see if this issue is claimed and if not, I'd love to take it on!

**Comment by StommePoes** [comment]
Seeing's how the contrast is still failing on the site when I visit, it looks like nobody has taken this on :P and I think folks will be thrilled that you have time/resources to tackle it!

**Comment by jlin95** [comment]
Hi! Is this issue still up to be worked on? And is @rachel-fischoff still keen on working on this? Would love to help out if possible :)

**Comment by ericwbailey** [comment]
Hi folks. We have a theme switcher now, so I'm considering this issue closed. 

Creating muted light and bright dark themes are beyond the scope of this issue, which is adding the ability to change the color theme. If we investigate those new themes, they should be a separate issue.
